### PR TITLE
r/sagemaker_model - add support for `image_config` + refactor tests

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #13826

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_sagemaker_model - add support for `primary_container. image_config` and `containers.image_config`
resource_aws_sagemaker_model - add plan time validation for `execution_role_arn`
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSSagemakerModel_'
--- PASS: TestAccAWSSagemakerModel_disappears (56.13s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerModeSingle (56.14s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerEnvironment (56.48s)
--- PASS: TestAccAWSSagemakerModel_basic (60.75s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerImageConfig (61.04s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerHostname (62.97s)
--- PASS: TestAccAWSSagemakerModel_networkIsolation (64.59s)
--- PASS: TestAccAWSSagemakerModel_containers (64.71s)
--- PASS: TestAccAWSSagemakerModel_vpcConfig (82.24s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerModelDataUrl (91.82s)
--- PASS: TestAccAWSSagemakerModel_tags (125.73s)
```

### CODE REVIEW

File: aws/resource_aws_sagemaker_model.go [hunk 1]
```diff
@@ -80,6 +79,20 @@ func resourceAwsSagemakerModel() *schema.Resource {
 							ValidateFunc: validateSagemakerEnvironment,
 							Elem:         &schema.Schema{Type: schema.TypeString},
 						},
+						"image_config": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_model.go:85]: Add `MaxItems: 1,`?

File: aws/resource_aws_sagemaker_model.go [hunk 2]
```diff
@@ -160,6 +172,20 @@ func resourceAwsSagemakerModel() *schema.Resource {
...
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_model.go:178]: Add `MaxItems: 1,`?

File: aws/resource_aws_sagemaker_model_test.go [hunk 3]
```diff
@@ -519,262 +529,186 @@ data "aws_iam_policy_document" "foo" {
...
+`, rName)
 }
 
-data "aws_iam_policy_document" "assume_role" {
-  statement {
-    actions = ["sts:AssumeRole"]
+func testAccSagemakerPrimaryContainerImageConfigConfig(rName string) string {
+	return testAccSagemakerModelConfigBase(rName) + fmt.Sprintf(`
+resource "aws_sagemaker_model" "test" {
+  name               = "%[1]q"
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_model_test.go:573]: Extra quotes.

```suggestion
  name               = %[1]q
```

**Review by ewbankkit — APPROVED** [review]
LGTM. Thanks.

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @DrFaust92 

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TESTARGS='-run=TestAccAWSSagemakerModel_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerModel_ -timeout 120m
=== RUN   TestAccAWSSagemakerModel_basic
=== PAUSE TestAccAWSSagemakerModel_basic
=== RUN   TestAccAWSSagemakerModel_tags
=== PAUSE TestAccAWSSagemakerModel_tags
=== RUN   TestAccAWSSagemakerModel_primaryContainerModelDataUrl
=== PAUSE TestAccAWSSagemakerModel_primaryContainerModelDataUrl
=== RUN   TestAccAWSSagemakerModel_primaryContainerHostname
=== PAUSE TestAccAWSSagemakerModel_primaryContainerHostname
=== RUN   TestAccAWSSagemakerModel_primaryContainerImageConfig
=== PAUSE TestAccAWSSagemakerModel_primaryContainerImageConfig
=== RUN   TestAccAWSSagemakerModel_primaryContainerEnvironment
=== PAUSE TestAccAWSSagemakerModel_primaryContainerEnvironment
=== RUN   TestAccAWSSagemakerModel_primaryContainerModeSingle
=== PAUSE TestAccAWSSagemakerModel_primaryContainerModeSingle
=== RUN   TestAccAWSSagemakerModel_containers
=== PAUSE TestAccAWSSagemakerModel_containers
=== RUN   TestAccAWSSagemakerModel_vpcConfig
=== PAUSE TestAccAWSSagemakerModel_vpcConfig
=== RUN   TestAccAWSSagemakerModel_networkIsolation
=== PAUSE TestAccAWSSagemakerModel_networkIsolation
=== RUN   TestAccAWSSagemakerModel_disappears
=== PAUSE TestAccAWSSagemakerModel_disappears
=== CONT  TestAccAWSSagemakerModel_basic
=== CONT  TestAccAWSSagemakerModel_primaryContainerModeSingle
=== CONT  TestAccAWSSagemakerModel_disappears
=== CONT  TestAccAWSSagemakerModel_containers
=== CONT  TestAccAWSSagemakerModel_vpcConfig
=== CONT  TestAccAWSSagemakerModel_primaryContainerHostname
=== CONT  TestAccAWSSagemakerModel_primaryContainerEnvironment
=== CONT  TestAccAWSSagemakerModel_primaryContainerImageConfig
=== CONT  TestAccAWSSagemakerModel_networkIsolation
=== CONT  TestAccAWSSagemakerModel_tags
=== CONT  TestAccAWSSagemakerModel_primaryContainerModelDataUrl
--- PASS: TestAccAWSSagemakerModel_basic (30.84s)
--- PASS: TestAccAWSSagemakerModel_disappears (31.40s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerEnvironment (33.58s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerModeSingle (33.92s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerImageConfig (35.67s)
--- PASS: TestAccAWSSagemakerModel_vpcConfig (38.47s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerModelDataUrl (40.47s)
2020/11/05 09:01:57 [DEBUG] Trying to get account information via sts:GetCallerIdentity
--- PASS: TestAccAWSSagemakerModel_primaryContainerHostname (45.46s)
--- PASS: TestAccAWSSagemakerModel_containers (46.30s)
--- PASS: TestAccAWSSagemakerModel_networkIsolation (46.62s)
--- PASS: TestAccAWSSagemakerModel_tags (58.80s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	60.302s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TESTARGS='-run=TestAccAWSSagemakerModel_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerModel_ -timeout 120m
=== RUN   TestAccAWSSagemakerModel_basic
=== PAUSE TestAccAWSSagemakerModel_basic
=== RUN   TestAccAWSSagemakerModel_tags
=== PAUSE TestAccAWSSagemakerModel_tags
=== RUN   TestAccAWSSagemakerModel_primaryContainerModelDataUrl
=== PAUSE TestAccAWSSagemakerModel_primaryContainerModelDataUrl
=== RUN   TestAccAWSSagemakerModel_primaryContainerHostname
=== PAUSE TestAccAWSSagemakerModel_primaryContainerHostname
=== RUN   TestAccAWSSagemakerModel_primaryContainerImageConfig
=== PAUSE TestAccAWSSagemakerModel_primaryContainerImageConfig
=== RUN   TestAccAWSSagemakerModel_primaryContainerEnvironment
=== PAUSE TestAccAWSSagemakerModel_primaryContainerEnvironment
=== RUN   TestAccAWSSagemakerModel_primaryContainerModeSingle
=== PAUSE TestAccAWSSagemakerModel_primaryContainerModeSingle
=== RUN   TestAccAWSSagemakerModel_containers
=== PAUSE TestAccAWSSagemakerModel_containers
=== RUN   TestAccAWSSagemakerModel_vpcConfig
=== PAUSE TestAccAWSSagemakerModel_vpcConfig
=== RUN   TestAccAWSSagemakerModel_networkIsolation
=== PAUSE TestAccAWSSagemakerModel_networkIsolation
=== RUN   TestAccAWSSagemakerModel_disappears
=== PAUSE TestAccAWSSagemakerModel_disappears
=== CONT  TestAccAWSSagemakerModel_basic
=== CONT  TestAccAWSSagemakerModel_primaryContainerModeSingle
=== CONT  TestAccAWSSagemakerModel_primaryContainerModelDataUrl
=== CONT  TestAccAWSSagemakerModel_networkIsolation
=== CONT  TestAccAWSSagemakerModel_disappears
=== CONT  TestAccAWSSagemakerModel_containers
=== CONT  TestAccAWSSagemakerModel_vpcConfig
=== CONT  TestAccAWSSagemakerModel_primaryContainerHostname
=== CONT  TestAccAWSSagemakerModel_primaryContainerEnvironment
=== CONT  TestAccAWSSagemakerModel_primaryContainerImageConfig
=== CONT  TestAccAWSSagemakerModel_tags
--- PASS: TestAccAWSSagemakerModel_containers (32.13s)
--- PASS: TestAccAWSSagemakerModel_basic (32.18s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerEnvironment (32.44s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerModelDataUrl (43.66s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerModeSingle (45.24s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerHostname (47.57s)
--- PASS: TestAccAWSSagemakerModel_disappears (49.25s)
--- PASS: TestAccAWSSagemakerModel_networkIsolation (50.78s)
--- PASS: TestAccAWSSagemakerModel_vpcConfig (53.50s)
--- PASS: TestAccAWSSagemakerModel_tags (58.82s)
--- PASS: TestAccAWSSagemakerModel_primaryContainerImageConfig (65.78s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	67.373s
```

**Pull request merged**